platform/darwin/src/MGLOfflineStorage.mm

```diff
@@ -40,58 +40,98 @@ + (instancetype)sharedOfflineStorage {
     return sharedOfflineStorage;
 }
 
+/**
+ Returns the file URL to the offline cache.
+
+ The cache is located in a directory specific to the application, so that packs
+ downloaded by other applications don’t count toward this application’s limits.
+
+ The cache is located at:
+ ~/Library/Application Support/tld.app.bundle.id/mapbox/cache.db
+ */
++ (NSURL *)cacheURL {
+    return [[self class] cacheURLIncludingSubdirectory:YES];
+}
+
+/**
+ Returns the file URL to the offline cache, with the option to omit the "mapbox"
+ subdirectory for legacy (v3.2.0 - v3.2.3) migration purposes.
+
+ The subdirectory-less cache was located at:
+ ~/Library/Application Support/tld.app.bundle.id/cache.db
+ */
++ (NSURL *)cacheURLIncludingSubdirectory:(BOOL)useSubdirectory {
+    NSURL *cacheDirectoryURL = [[NSFileManager defaultManager] URLForDirectory:NSApplicationSupportDirectory
+                                                                      inDomain:NSUserDomainMask
+                                                             appropriateForURL:nil
+                                                                        create:YES
+                                                                         error:nil];
+    NSString *bundleIdentifier = [NSBundle mainBundle].bundleIdentifier;
+    if (!bundleIdentifier) {
+        // There’s no main bundle identifier when running in a unit test bundle.
+        bundleIdentifier = [NSBundle bundleForClass:self].bundleIdentifier;
+    }
+    cacheDirectoryURL = [cacheDirectoryURL URLByAppendingPathComponent:bundleIdentifier];
+    if (useSubdirectory) {
+        cacheDirectoryURL = [cacheDirectoryURL URLByAppendingPathComponent:@"mapbox"];
```

Comment:
I'd go with something even less likely to conflict with developer code, such as `.mapbox`. For that matter, I wonder if dot-directories are even backed up to iCloud or Time Machine in the first place.